### DISCUSSION TITLE
Spacy NER data in required format

### DISCUSSION THREAD

**Discussion by hiteshpara** [explosion/spaCy#9896] [Help: Coding & Implementations] [answered]
I have data in csv format where I have text, keyphrase, label, start, and end index.
IS there any possible way spacy that it automatically converts this data into spacy required NER format

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @hiteshpara , when training a spaCy model, it needs to be saved [into a serialized spaCy format](https://spacy.io/api/docbin). This will allow you to efficiently store huge amounts of data. Since you already have the `label`, `start`, `end`, and `text`, it should be straightforward. You can check the [NER Demo Project](https://github.com/explosion/projects/tree/v3/pipelines/ner_demo) (especially the [`convert.py`](https://github.com/explosion/projects/blob/v3/pipelines/ner_demo/scripts/convert.py)) as a guide.
